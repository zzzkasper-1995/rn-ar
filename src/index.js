import {registerComponent, Navigation} from './core/navigation';
import {OnboardingScreen, PlaygroundScreen, IndicatorsScreen, MainScreen} from './modules';
import {AppScreen} from './core/app';
import {Offline, Fab, BlurOverlay} from './simple';
import {settingsDefault, rootLoadApp} from './routes';
import {ArSceneScreen} from './modules/ArScene';

/** Инициализация модулей */
function initModules() {
	// простые компоненты
	registerComponent('blurOverlay', BlurOverlay); // Слой размытия для режима в фоне
	registerComponent('fab', Fab); // фаб меню
	registerComponent('offline', Offline); // планка оффлайн режима
	// умные компоненты
	registerComponent('main', MainScreen);
	registerComponent('indicators', IndicatorsScreen);
	registerComponent('playground', PlaygroundScreen);
	registerComponent('arScene', ArSceneScreen);
	registerComponent('onboarding', OnboardingScreen);
	registerComponent('initApp', AppScreen);
}

/** Инициализация дерева навигаци */
function initRoutes() {
	Navigation.events().registerAppLaunchedListener(() => {
		Navigation.setDefaultOptions(settingsDefault);
		Navigation.setRoot(rootLoadApp);
	});
}

/** Точка входа */
export default function runApp() {
	initModules();
	initRoutes();
}
