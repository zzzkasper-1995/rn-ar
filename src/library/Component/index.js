import BindComponent from './full';
import BindSimple from './simple';

export {BindComponent, BindSimple};
