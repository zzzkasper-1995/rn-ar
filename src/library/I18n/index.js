import React from 'react';
import {Text} from 'react-native';
import i18n from './i18n';

const db = {};
const getText = key => {
	const text = i18n.t(key, {language: i18n.currentLocale()});
	if (i18n.t(key, {language: i18n.currentLocale()}).includes('[missing')) {
		db[i18n.currentLocale()] = {...db[i18n.currentLocale()], [key]: key};
		return key;
	}
	return text;
};
const I = ({style = {}, text = 'default'}) => <Text style={style}>{getText(text)}</Text>;

I.text = (text = 'default') => getText(text);

I.printNotFound = () => {
	// какой либо запрос или просто вывод
	console.log('Not found localization', db);
};

export {I};
