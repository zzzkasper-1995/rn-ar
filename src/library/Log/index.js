import Options from '../../options';

const Log = (...params) => {
	if (Options.isLog && __DEV__ === true) {
		console.log(...params);
	}
};

export {Log};
