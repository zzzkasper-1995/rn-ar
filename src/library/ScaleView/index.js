import React from 'react';
import {Animated} from 'react-native';

const DURATION = 200;

class ScaleView extends React.PureComponent {
	constructor(props) {
		super(props);
		this.scaleValue = new Animated.Value(1);
	}

	animateBig = () => {
		Animated.timing(this.scaleValue, {
			toValue: 0.95,
			duration: DURATION,
			useNativeDriver: true,
		}).start();
	};

	animateLitle = () => {
		Animated.timing(this.scaleValue, {
			toValue: 1,
			duration: DURATION,
			useNativeDriver: true,
		}).start();
	};

	render() {
		const {style, render, children, ...props} = this.props;

		const scale = {
			transform: [
				{
					scale: this.scaleValue,
				},
			],
		};

		const touchProps = {
			onPressIn: this.animateBig,
			onPressOut: this.animateLitle,
		};

		return (
			<Animated.View style={[scale, style, {flex: 1}]} {...props}>
				{render && render(touchProps)}
			</Animated.View>
		);
	}
}

export {ScaleView};
