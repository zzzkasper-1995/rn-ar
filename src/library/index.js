export * from './BackHandler';
export * from './Component';
export * from './I18n';
export * from './Links';
export * from './Log';
export * from './ScaleView';
export * from './StatusBar';
export * from './Theme';
export * from './Toast';
export * from './TouchID';
export * from './UI';
export * from './Utils';
export * from './Validates';
