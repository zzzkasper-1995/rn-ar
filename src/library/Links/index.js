import {Linking} from 'react-native';
import {Log} from '../Log';

/**
 * Обертка над линкс. Открывает ссылку
 * @param {String} url  ссылка для открытия
 */
const onLinking = url => {
	console.log('onLinking', url);

	Linking.canOpenURL(url)
		.then(supported => {
			if (!supported) {
				// Log('onLinking error');
			} else {
				return Linking.openURL(url);
			}
		})
		.catch(err => Log('onLinking error:', err));
};

export const Links = {
	onLinking,
};
