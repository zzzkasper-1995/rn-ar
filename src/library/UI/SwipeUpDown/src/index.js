// https://github.com/react-native-vietnam/react-native-swipe-up-down
import React, {Component} from 'react';
import {
	Platform,
	StyleSheet,
	View,
	PanResponder,
	Dimensions,
	LayoutAnimation,
	TouchableOpacity,
} from 'react-native';

import SwipeIcon from './components/SwipeIcon';
import images from './assets/images';
import {Log} from '../../../Log';

const MARGIN_TOP = Platform.OS === 'ios' ? 20 : 0;
const DEVICE_HEIGHT = Dimensions.get('window').height - MARGIN_TOP;
type Props = {
	hasRef?: () => void,
	swipeHeight?: number,
	itemMini?: Function,
	itemFull: Function,
	disablePressToShow?: boolean,
	style?: object,
	onShowMini?: () => void,
	onShowFull?: () => void,
	animation?: 'linear' | 'spring' | 'easeInEaseOut' | 'none',
	topDragHeight?: number,
	initCollapsed?: Boolean,
};

export default class SwipeUpDown extends Component<Props> {
	static defautProps = {
		disablePressToShow: false,
		topDragHeight: 0,
		itemMini: undefined,
		initCollapsed: true,
	};

	constructor(props) {
		super(props);
		this.state = {
			collapsed: props.initCollapsed,
		};
		this.disablePressToShow = props.disablePressToShow;
		this.SWIPE_HEIGHT = props.swipeHeight || 60;
		this._panResponder = null;
		this.top = this.SWIPE_HEIGHT;
		this.height = this.SWIPE_HEIGHT;
		this.customStyle = {
			style: {
				bottom: 0,
				top: this.top,
				height: this.height,
			},
		};
		this.checkCollapsed = true;
		this.showFull = this.showFull.bind(this);
	}

	componentWillMount() {
		this._panResponder = PanResponder.create({
			onMoveShouldSetPanResponder: (event, gestureState) => true,
			onPanResponderMove: this._onPanResponderMove.bind(this),
			onPanResponderRelease: this._onPanResponderRelease.bind(this),
		});
	}

	componentDidMount() {
		this.props.hasRef && this.props.hasRef(this);
	}

	getPan = () => {
		return this._panResponder.panHandlers;
	};

	showFull() {
		const {onShowFull} = this.props;
		this.customStyle.style.top = 0;
		this.customStyle.style.height = DEVICE_HEIGHT;
		this.swipeIconRef && this.swipeIconRef.setState({icon: images.arrow_down, showIcon: true});
		this.updateNativeProps();
		this.state.collapsed && this.setState({collapsed: false});
		this.checkCollapsed = false;
		onShowFull && onShowFull();
	}

	showMini() {
		const {onShowMini, itemMini} = this.props;
		this.customStyle.style.top = itemMini ? DEVICE_HEIGHT - this.SWIPE_HEIGHT : DEVICE_HEIGHT;
		this.customStyle.style.height = itemMini ? this.SWIPE_HEIGHT : 0;
		this.swipeIconRef && this.swipeIconRef.setState({showIcon: false});
		this.updateNativeProps();
		!this.state.collapsed && this.setState({collapsed: true});
		this.checkCollapsed = true;
		onShowMini && onShowMini();
	}

	_onPanResponderRelease(event, gestureState) {
		if (gestureState.dy < -100 || gestureState.dy < 100) {
			this.showFull();
		} else {
			this.showMini();
		}
	}

	_onPanResponderMove(event, gestureState) {
		if (gestureState.dy > 0 && !this.checkCollapsed) {
			// SWIPE DOWN

			this.customStyle.style.top = this.top + gestureState.dy;
			this.customStyle.style.height = DEVICE_HEIGHT - gestureState.dy;
			this.swipeIconRef && this.swipeIconRef.setState({icon: images.minus});
			// !this.state.collapsed && this.setState({ collapsed: true });
			this.updateNativeProps();
		} else if (this.checkCollapsed && gestureState.dy < -60) {
			// SWIPE UP
			this.top = 0;
			this.customStyle.style.top = DEVICE_HEIGHT + gestureState.dy;
			this.customStyle.style.height = -gestureState.dy + this.SWIPE_HEIGHT;
			this.swipeIconRef && this.swipeIconRef.setState({icon: images.minus, showIcon: true});
			if (this.customStyle.style.top <= DEVICE_HEIGHT / 2) {
				this.swipeIconRef &&
					this.swipeIconRef.setState({
						icon: images.arrow_down,
						showIcon: true,
					});
			}
			this.updateNativeProps();
			this.state.collapsed && this.setState({collapsed: false});
		}
	}

	updateNativeProps() {
		switch (this.props.animation) {
			case 'linear':
				LayoutAnimation.linear();
				break;
			case 'spring':
				LayoutAnimation.spring();
				break;
			case 'easeInEaseOut':
				LayoutAnimation.easeInEaseOut();
				break;
			case 'none':
			default:
				break;
		}
		this.viewRef.setNativeProps(this.customStyle);
	}

	render() {
		// Log('SwipeUpDown', this.state, this.props);
		const {itemMini, itemFull, style, topDragHeight} = this.props;
		const {collapsed} = this.state;
		return (
			<View
				ref={ref => (this.viewRef = ref)}
				style={[
					styles.wrapSwipe,
					{
						height: this.SWIPE_HEIGHT,
						marginTop: MARGIN_TOP,
						backgroundColor: 'red',
					},
					!itemMini && collapsed && {marginBottom: -200},
					style,
				]}
			>
				<TouchableOpacity
					style={{height: topDragHeight}}
					{...this._panResponder.panHandlers}
					onPressOut={() => this.showMini()}
				/>
				<SwipeIcon onClose={() => this.showMini()} hasRef={ref => (this.swipeIconRef = ref)} />
				{collapsed ? (
					<TouchableOpacity
						activeOpacity={this.disablePressToShow ? 1 : 0.6}
						style={{height: this.SWIPE_HEIGHT}}
						onPress={() => !this.disablePressToShow && this.showFull()}
					>
						{itemMini && itemMini()}
					</TouchableOpacity>
				) : (
					itemFull(this._panResponder.panHandlers)
				)}
			</View>
		);
	}
}

const styles = StyleSheet.create({
	wrapSwipe: {
		position: 'absolute',
		bottom: 0,
		left: 0,
		right: 0,
		top: 0,
	},
});
