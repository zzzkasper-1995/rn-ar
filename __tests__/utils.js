/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Utils, Text, View, Button} from '../src/library';
// import {Text} from '../src/library';

// Note: test renderer must be required after react-native.

test('renders correctly text', () => {
	const tree = renderer.create(<Text />).toJSON();
	expect(tree).toMatchSnapshot();
});

test('renders correctly view', () => {
	const tree = renderer.create(<View />).toJSON();
	expect(tree).toMatchSnapshot();
});

test('renders correctly button', () => {
	const tree = renderer.create(<Button />).toJSON();
	expect(tree).toMatchSnapshot();
});

test(' correctly rounding', () => {
	let value = Utils.rounding(0.0001, 1);
	expect(value).toBe('0');
	value = Utils.rounding(0.000323232323321, 1);
	expect(value).toBe('0');
	value = Utils.rounding(21, 1);
	expect(value).toBe('21');
	value = Utils.rounding(2324234.0001, 1);
	expect(value).toBe('2 324 234');
	value = Utils.rounding(23424324324234324.002342343201, 150);
	expect(value).toBe('23 424 324 324 234 300');
});

test('time rounding 1 000 000', () => {
	const ar = [];
	const length = 1000000;
	for (let i = 0; i < length; i += 1) {
		const v = 10 * i * Math.random() ** (Math.random() * 100);
		ar.push({value: v, eps: Math.random()});
	}

	ar.forEach(el => {
		Utils.rounding(el.value, el.eps);
	});
});
