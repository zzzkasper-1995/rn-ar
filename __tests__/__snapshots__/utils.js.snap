// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "rgb(51, 51, 51)",
      "fontSize": 16,
    }
  }
/>
`;

exports[`renders correctly button 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(255,255,255)",
      "stylesView": Object {
        "backgroundColor": "rgb(51, 51, 51)",
        "weight": 400,
      },
    }
  }
>
  <Text
    i18n={true}
    style={
      Object {
        "color": undefined,
        "fontSize": 16,
      }
    }
  >
    text not found
  </Text>
</View>
`;

exports[`renders correctly rounding 1`] = `"0"`;

exports[`renders correctly text 1`] = `
<Text
  style={
    Object {
      "color": "rgb(51, 51, 51)",
      "fontSize": 16,
    }
  }
/>
`;

exports[`renders correctly view 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(255,255,255)",
    }
  }
/>
`;
